src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -237,6 +238,10 @@ void Bank::read_and_add_tensor(std::istream& stream, int64_t uid, const std::str
     ov::Tensor allocated_tensor;
 
     // FIXME: reading not via a dedicated function
+    bool is_intialized = false;
+    read(stream, is_intialized);
+    NPUW_ASSERT(is_intialized);
```

Comment:
this too. Haven't you wrote empty tensors before?
if `is_initialized` must always be `true`, why write it to the stream at all?

Reply:
I changed Tensor serialization in some of the previous PRs. It occurred when I was serializing empty scales/zerops. This change is just to align with the previous change

Reply:
I don't get it. Do you serialize empty tensors now or not?
If you don't, you don't need this `bool` in the stream either.

If you do, this code breaks.

Reply:
I do but not for the bank

Reply:
did I read this right you have a common routine to write a tensor to disk but you read it here in some specific way?
it doesn't sound good tbh.

you'd need to use the same function to read tensors as you write those. If that routine checks for `false` and does nothing, but you still need to raise an assert, you can make it in the caller code (= HERE) if the resulting tensor is still empty after read.

Reply:
Discussed on call, this change is caused by optimization (reading from stream directly to the L0 buffer)

Reply:
Discussed locally